code/controllers/subsystem/bluespace_exploration.dm

```diff
@@ -0,0 +1,483 @@
+#define CLEAR_TURF_PROCESSING_TIME 600
+
+SUBSYSTEM_DEF(bluespace_exploration)
+	name = "Bluespace Exploration"
+	wait = 1
+	priority = FIRE_PRIORITY_EXPLORATION
+	init_order = INIT_ORDER_BS_EXPLORATION
+
+	var/generating = FALSE
+	var/generating_level
+
+	//Which systems are ours?
+	var/list/bluespace_systems	// Key = /datum/space_level, Value = Boolean (Are we in use)
+
+	//Starmap generation
+	var/datum/star_system/current_system = null
+	var/list/star_systems
+	var/list/star_links
+
+	//Ruin generation
+	var/list/ruin_templates
+	var/obj/docking_port/stationary/away_mission_port
+
+	//=====Factions=====
+	// factions[datum] = new datum
+	var/list/factions
+
+	//=====Ship Tracking=====
+	var/list/spawnable_ships
+	//All ships that are able to use the random generation must be tracked
+	var/list/tracked_ships
+
+	//=====Ship Processing Queue=====
+	// ! Tracks a list of all the ships requesting transit
+	// ! Key (string) - shuttle_id
+	// ! Value (/datum/data_holder/bluespace_exploration) - generation settings
+	var/list/ship_traffic_queue	//The queue for generation is disguised as 'bluespace traffic control'
+
+	//=====Z-Level Wiping=====
+	//Are we currently processing a z-level wipe
+	var/wiping_z_level = FALSE
+	//The target z-level of the wipe
+	var/target_z_level = 0
+	//A list of all turfs on the z-leves divided into equal parts depending on how many ticks we have to wipe
+	var/list/wiping_divided_turfs
+	//The turf group we are currently wiping
+	var/wipe_process_num  = 0
+	//Data holder for turf wiping
+	var/datum/data_holder/bluespace_exploration/wipe_data_holder
+	//The queue for z-levels to wipe
+	var/list/z_level_queue
+
+/datum/controller/subsystem/bluespace_exploration/New()
+	. = ..()
+	//Hello lists (Maximum internal arrays exceeded need for this)
+	//Also this has to be done before SSmapping init
+	bluespace_systems = list()
+	z_level_queue = list()
+	ship_traffic_queue = list()
+	tracked_ships = list()
+	spawnable_ships = list()
+	factions = list()
+	star_systems = list()
+	ruin_templates = list()
+	star_links = list()
+
+/datum/controller/subsystem/bluespace_exploration/Initialize(start_timeofday)
+	. = ..()
+	//Create factions
+	for(var/faction_datum in subtypesof(/datum/faction))
+		factions[faction_datum] = new faction_datum
+	//Create z-levels
+	//Todo: Check low mem mode
+	for(var/i in 1 to CONFIG_GET(number/bluespace_exploration_levels))
+		bluespace_systems[SSmapping.add_new_zlevel("Bluespace Exploration Level [i]", ZTRAITS_BLUESPACE_EXPLORATION)] = FALSE
+
+/datum/controller/subsystem/bluespace_exploration/fire(resumed = 0)
```

Comment:
You should break this up the way SSair does with `currentpart` (see `air.dm`). Also, you should `if(MC_TICK_CHECK)` rather than `CHECK_TICK`.